File: backend/app/models/telemetry.py
from pydantic import BaseModel


class TelemetryRow(BaseModel):
    timestamp: int
    temperature: float
    pressure: float
    voltage: float
    battery: float


class AnomalyRow(TelemetryRow):
    sensor: str


class SensorSummary(BaseModel):
    sensor: str
    min_value: float
    max_value: float
    mean_value: float
    std_dev: float

File: frontend/src/types/telemetry.ts
export interface TelemetryRow {
    timestamp: number;
    temperature: number;
    pressure: number;
    voltage: number;
    battery: number;
}

export interface AnomalyRow extends TelemetryRow {
    sensor: string;
}

export interface SensorSummary {
    sensor: string;
    min_value: number;
    max_value: number;
    mean_value: number;
    std_dev: number;
}